include/ap/argument_parser.hpp

```diff
@@ -211,6 +233,12 @@ class optional_argument : public detail::argument_interface {
         return *this;
     }
 
+    inline optional_argument& choices(const std::vector<value_type>& choices)
+    requires(utility::equality_comparable<value_type>) {
+        this->_choices = choices;
+        return *this;
+    }
+
     optional_argument& default_value(const std::any& default_value) {
```

Comment:
Add valid choice checking to the default_value method (add tests covering invalid default value)